### ISSUE TITLE
`mutate` can't perform sums on grouped dataframes: `error in 'list_unchop()': '...' must be empty.`

### ISSUE THREAD

**Issue by andrewbaxter439** [tidyverse/dplyr#6537]
This appears to have cropped up in latest update or `dplyr` on GitHub (1.0.99.9000). Seems to be passing an incorrect argument to `vctrs::list_unchop`?

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

tibble(a = c("A", "A", "B", "B"),
       b = sample(100, 4)) |> 
  group_by(a) |> 
  mutate(diff = diff(b))
#> Error in `mutate()`:
#> ℹ In argument: `diff = diff(b)`.
#> ℹ In group 2: `a = "B"`.
#> Caused by error in `list_unchop()`:
#> ! `...` must be empty.
#> ✖ Problematic argument:
#> • ..1 = rows
#> ℹ Did you forget to name an argument?

#> Backtrace:
#>      ▆
#>   1. ├─dplyr::mutate(...)
#>   2. └─dplyr:::mutate.data.frame(...)
#>   3.   └─dplyr:::mutate_cols(.data, dplyr_quosures(...))
#>   4.     ├─base::withCallingHandlers(...)
#>   5.     └─dplyr:::mutate_col(dots[[i]], .data, mask, new_columns)
#>   6.       └─vctrs::list_unchop(chunks, rows)
#>   7.         └─rlang::check_dots_empty0(...)
#>   8.           └─rlang::check_dots_empty(call = call)
#>   9.             └─rlang:::action_dots(...)
#>  10.               ├─base (local) try_dots(...)
#>  11.               └─rlang (local) action(...)


tibble(a = c("A", "B"),
       b1 = sample(100, 2),
       b2 = sample(100, 2)) |> 
  group_by(a) |> 
  mutate(y = b1 + b2)
#> Error in `mutate()`:
#> ℹ In argument: `y = b1 + b2`.
#> ℹ In group 2: `a = "B"`.
#> Caused by error in `list_unchop()`:
#> ! `...` must be empty.
#> ✖ Problematic argument:
#> • ..1 = rows
#> ℹ Did you forget to name an argument?

#> Backtrace:
#>      ▆
#>   1. ├─dplyr::mutate(...)
#>   2. └─dplyr:::mutate.data.frame(...)
#>   3.   └─dplyr:::mutate_cols(.data, dplyr_quosures(...))
#>   4.     ├─base::withCallingHandlers(...)
#>   5.     └─dplyr:::mutate_col(dots[[i]], .data, mask, new_columns)
#>   6.       └─vctrs::list_unchop(chunks, rows)
#>   7.         └─rlang::check_dots_empty0(...)
#>   8.           └─rlang::check_dots_empty(call = call)
#>   9.             └─rlang:::action_dots(...)
#>  10.               ├─base (local) try_dots(...)
#>  11.               └─rlang (local) action(...)
```

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31 ucrt)
#>  os       Windows 10 x64 (build 19044)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  English_United Kingdom.utf8
#>  ctype    English_United Kingdom.utf8
#>  tz       Europe/London
#>  date     2022-11-14
#>  pandoc   2.18 @ C:/Program Files/RStudio/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version     date (UTC) lib source
#>  cli           3.4.1       2022-09-23 [1] CRAN (R 4.2.1)
#>  digest        0.6.30      2022-10-18 [1] CRAN (R 4.2.2)
#>  dplyr       * 1.0.99.9000 2022-10-04 [1] Github (tidyverse/dplyr@e2770ef)
#>  evaluate      0.18        2022-11-07 [1] CRAN (R 4.2.2)
#>  fansi         1.0.3       2022-03-24 [1] CRAN (R 4.2.0)
#>  fastmap       1.1.0       2021-01-25 [1] CRAN (R 4.1.0)
#>  fs            1.5.2       2021-12-08 [1] CRAN (R 4.1.2)
#>  generics      0.1.3       2022-07-05 [1] CRAN (R 4.2.1)
#>  glue          1.6.2       2022-02-24 [1] CRAN (R 4.2.0)
#>  highr         0.9         2021-04-16 [1] CRAN (R 4.1.0)
#>  htmltools     0.5.3       2022-07-18 [1] CRAN (R 4.2.1)
#>  knitr         1.40        2022-08-24 [1] CRAN (R 4.2.1)
#>  lifecycle     1.0.3       2022-10-07 [1] CRAN (R 4.2.2)
#>  magrittr      2.0.3       2022-03-30 [1] CRAN (R 4.2.0)
#>  pillar        1.8.1       2022-08-19 [1] CRAN (R 4.2.1)
#>  pkgconfig     2.0.3       2019-09-22 [1] CRAN (R 4.1.0)
#>  purrr         0.3.5       2022-10-06 [1] CRAN (R 4.2.2)
#>  R.cache       0.16.0      2022-07-21 [1] CRAN (R 4.2.1)
#>  R.methodsS3   1.8.2       2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo          1.25.0      2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils       2.12.2      2022-11-11 [1] CRAN (R 4.2.2)
#>  R6            2.5.1       2021-08-19 [1] CRAN (R 4.1.1)
#>  reprex        2.0.2       2022-08-17 [1] CRAN (R 4.2.1)
#>  rlang         1.0.6       2022-09-24 [1] CRAN (R 4.2.1)
#>  rmarkdown     2.18        2022-11-09 [1] CRAN (R 4.2.2)
#>  rstudioapi    0.14        2022-08-22 [1] CRAN (R 4.2.1)
#>  sessioninfo   1.2.2       2021-12-06 [1] CRAN (R 4.1.2)
#>  stringi       1.7.8       2022-07-11 [1] CRAN (R 4.2.1)
#>  stringr       1.4.1       2022-08-20 [1] CRAN (R 4.2.1)
#>  styler        1.8.1       2022-11-07 [1] CRAN (R 4.2.2)
#>  tibble        3.1.8       2022-07-22 [1] CRAN (R 4.2.1)
#>  tidyselect    1.2.0       2022-10-10 [1] CRAN (R 4.2.2)
#>  utf8          1.2.2       2021-07-24 [1] CRAN (R 4.1.1)
#>  vctrs         0.5.0       2022-10-22 [1] CRAN (R 4.2.2)
#>  withr         2.5.0       2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun          0.34        2022-10-18 [1] CRAN (R 4.2.2)
#>  yaml          2.3.6       2022-10-18 [1] CRAN (R 4.2.1)
#> 
#>  [1] C:/R Packages
#>  [2] C:/Program Files/R/R-4.2.2/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by andrewbaxter439** [comment]
Update: having reinstalled dplyr it's working again by the latest commit. The full tidyverse installation seemed to install e2770ef which has the error. Updating to latest version fixes it. Maybe an alteration to the tidyverse installation process rather than `dplyr`. Happy to close if not resolvable here.

**Comment by hadley** [comment]
How did you install it?

**Comment by andrewbaxter439** [comment]
Updated R to 4.2.2 then `update.packages()`. Can't reproduce in fresh environment, except by specifying installing that commit of `dplyr`. This may have been a very odd one-off.

**Comment by hadley** [comment]
You might've needed to restart R.

**Comment by andrewbaxter439** [comment]
Ran same steps on other computer, updated `dplyr` to 1a7061f this time which gave the same problem (both before and after restarting R). Fixed by installing 4ce9675 (updating to CRAN `vctrs`). Seems `update.packages` sometimes selects an unpredictable source commit to compile from? I imagine this would be a very rare problem irl, happy to close.

**Comment by hadley** [comment]
As far as I know `update.packages()` will not update from github, so I don't know what's going on for you.
